src/EnergyPlus/HVACMultiSpeedHeatPump.cc

```diff
@@ -718,102 +639,72 @@ namespace HVACMultiSpeedHeatPump {
-                }
-                GetFanIndex(state, Alphas(7), MSHeatPump(MSHPNum).FanNum, ErrorsFound, state.dataHVACMultiSpdHP->CurrentModuleObject);
-                GetFanType(state, Alphas(7), FanType, ErrorsFound);
-                if (FanType != MSHeatPump(MSHPNum).FanType) {
-                    ShowSevereError(state,
-                                    format("{}, \"{}\", {} and {} do not match in Fan objects.",
-                                           state.dataHVACMultiSpdHP->CurrentModuleObject,
-                                           MSHeatPump(MSHPNum).Name,
-                                           cAlphaFields(6),
-                                           cAlphaFields(7)));
-                    ShowContinueError(state, format("The entered {} = {} and {} = {}", cAlphaFields(7), Alphas(7), cAlphaFields(6), Alphas(6)));
-                    ErrorsFound = true;
-                }
+            bool errFound = false;
+            Fans::GetFanIndex(state, Alphas(7), thisMSHP.FanNum, errFound, state.dataHVACMultiSpdHP->CurrentModuleObject);
+            if (!errFound && thisMSHP.FanNum > 0) {
+                thisMSHP.FanName = state.dataFans->Fan(thisMSHP.FanNum).FanName;
+                thisMSHP.FanType = state.dataFans->Fan(thisMSHP.FanNum).FanType_Num;
+                thisMSHP.FanInletNode = state.dataFans->Fan(thisMSHP.FanNum).InletNodeNum;
+                thisMSHP.FanOutletNode = state.dataFans->Fan(thisMSHP.FanNum).OutletNodeNum;
+                BranchNodeConnections::SetUpCompSets(state,
+                                                     state.dataHVACMultiSpdHP->CurrentModuleObject,
+                                                     thisMSHP.Name,
+                                                     DataHVACGlobals::cFanTypes(thisMSHP.FanType),
+                                                     thisMSHP.FanName,
+                                                     "UNDEFINED",
+                                                     "UNDEFINED");
             } else {
-                ShowSevereError(state,
-                                format("{}, \"{}\", {} is not allowed = {}",
-                                       state.dataHVACMultiSpdHP->CurrentModuleObject,
-                                       MSHeatPump(MSHPNum).Name,
-                                       cAlphaFields(6),
-                                       Alphas(6)));
-                ShowContinueError(state, "Valid choices are Fan:OnOff or Fan:ConstantVolume");
+                ShowSevereError(
+                    state,
+                    format(
+                        "{}, \"{}\", fan {} \"{}\" not found", state.dataHVACMultiSpdHP->CurrentModuleObject, thisMSHP.Name, Alphas(6), Alphas(7)));
+                ShowContinueError(state, "Correct this fan name using valid Fan:OnOff or Fan:ConstantVolume fan object names.");
```

Comment:
New warning (1st from GetFanIndex call and 2nd from MSHP here):

    ** Severe  ** AirLoopHVAC:UnitaryHeatPump:AirToAir:MultiSpeed, GetFanIndex: Fan not found=SUPPLY FAN 1A
    ** Severe  ** AirLoopHVAC:UnitaryHeatPump:AirToAir:MultiSpeed, "DXAC HEAT PUMP 1", fan FAN:ONOFF "SUPPLY FAN 1A" not found
    **   ~~~   ** Correct this fan name using valid Fan:OnOff or Fan:ConstantVolume fan object names.